test/unit/catalog-component.spec.ts

```diff
@@ -0,0 +1,181 @@
+
+let catalogComponent: ICatalogComponent
+let dappsDatabase: AppComponents['dappsDatabase']
+let picks: IPicksComponent
+let dbClientQueryMock: jest.Mock
+let dbClientReleaseMock: jest.Mock
+let segmentWriteKey: string
+let analyticsTrackMock: jest.Mock
+
+beforeEach(async () => {
+  analyticsTrackMock = jest.fn()
+  dbClientQueryMock = jest.fn()
+  dbClientReleaseMock = jest.fn().mockResolvedValue(undefined)
+  dappsDatabase = createTestPgComponent({
+    getPool: jest.fn().mockReturnValue({
+      connect: () => ({
+        query: dbClientQueryMock,
+        release: dbClientReleaseMock
+      })
+    })
+  })
+
+  picks = {
+    getPicksStats: jest.fn(),
+    getPicksByItemId: jest.fn(),
+    pickAndUnpickInBulk: jest.fn()
+  }
+
+  segmentWriteKey = 'testSegmentWriteKey'
+  catalogComponent = await createCatalogComponent({ dappsDatabase, picks }, segmentWriteKey)
+})
+
+describe('Catalog Component', () => {
+  describe('fetch', () => {
+    describe('and there are no results from the search text query', () => {
+      beforeEach(() => {
+        dbClientQueryMock.mockResolvedValueOnce({ rows: [] })
+      })
+
+      it('should return empty data if no items match the search text', async () => {
+        const filters = { search: 'nonexistent item' }
+        const result = await catalogComponent.fetch(filters, { searchId: 'testSearchId', anonId: 'anon123' })
+
+        expect(result).toEqual({ data: [], total: 0 })
+        expect(dbClientQueryMock).toHaveBeenCalledTimes(1)
+      })
+    })
+
+    describe('and there are results from the search text query', () => {
+      const items = [
+        {
+          id: 'item1',
+          name: 'Item 1',
+          item_type: FragmentItemType.WEARABLE_V1,
+          image:
+            'https://peer.decentraland.zone/lambdas/collections/contents/urn:decentraland:amoy:collections-v2:0x60aba8c01848c7ef7c1a5451ff11ca05b2b4893e:0/thumbnail'
+        },
+        {
+          id: 'item2',
+          name: 'Item 2',
+          item_type: FragmentItemType.WEARABLE_V1,
+          image:
+            'https://peer.decentraland.zone/lambdas/collections/contents/urn:decentraland:amoy:collections-v2:0x60aba8c01848c7ef7c1a5451ff11ca05b2b4893e:0/thumbnail'
+        }
+      ]
+      beforeEach(() => {
+        dbClientQueryMock.mockResolvedValueOnce({
+          rows: [
+            { id: 'item1', word: 'Item 1', word_similarity: 0.9, match_type: 'exact' },
+            { id: 'item2', word: 'Item 2', word_similarity: 0.8, match_type: 'partial' }
+          ]
+        })
+        dbClientQueryMock.mockResolvedValueOnce({
+          rows: items.map(item => ({ ...item, total: items.length }))
+        })
+        ;(Analytics as jest.Mock).mockImplementation(() => {
+          return {
+            track: analyticsTrackMock
+          }
+        })
+      })
+
+      it('should return catalog items based on the provided filters', async () => {
+        ;(picks.getPicksStats as jest.Mock).mockResolvedValue([])
```

Comment:
This seems to be repeated in both tests, would you mind moving it to the `beforeEach`?